Sources/StreamChat/Models/ChannelType.swift

```diff
@@ -49,18 +53,34 @@ public enum ChannelType: Codable, Hashable {
         case "team": self = .team
         case "gaming": self = .gaming
         case "commerce": self = .commerce
-        default: self = .custom(rawValue)
+        default:
+            Self.assertCustomTypeValue(rawValue)
+            self = .custom(rawValue)
         }
     }
     
     public init(from decoder: Decoder) throws {
         let container = try decoder.singleValueContainer()
         let value = try container.decode(String.self)
         self.init(rawValue: value)
+        
+        if case let .custom(value) = self {
+            Self.assertCustomTypeValue(value)
+        }
     }
     
     public func encode(to encoder: Encoder) throws {
         var container = encoder.singleValueContainer()
         try container.encode(rawValue)
     }
+    
+    /// Check that custom types have valid `rawValue`
+    private static func assertCustomTypeValue(_ value: String) {
+        let allowedCharacters = CharacterSet.lowercaseLetters.union(CharacterSet(charactersIn: "0123456789_"))
+        let valueCharacters = CharacterSet(charactersIn: value)
+        log.assert(
+            valueCharacters.isSubset(of: allowedCharacters),
+            "Allowed characters for `ChannelType.custom` are small letters, numbers and underscore."
```

Comment:
nit: Can we print the actual value in the error message? It will help with debugging